src/app/zap-templates/templates/chip/CHIPClusters-src.zapt

```diff
@@ -21,8 +21,34 @@ CHIP_ERROR {{asCamelCased clusterName false}}Cluster::{{asCamelCased name false}
 #ifdef CHIP_APP_USE_INTERACTION_MODEL
     VerifyOrReturnError(mDevice != nullptr, CHIP_ERROR_INCORRECT_STATE);
     (void) onCompletion;
-    // TODO(@vivien-apple): onCompletion is not used by IM for now.
-    chip::app::cluster::{{asCamelCased clusterName false}}::Encode{{asType name}}Command(mDevice->GetCommandSender(), mEndpoint, 0{{#chip_server_cluster_command_arguments}}, {{asCamelCased label}}{{/chip_server_cluster_command_arguments}});
+
+    Command::CommandParams cmdParams = { ZCLendpointId, ZCLgroupId, {{asCamelCased clusterName false}}::kClusterId, {{asCamelCased clusterName false}}::k{{asCamelCased name false}}CommandId,
+                                         (chip::app::Command::kCommandPathFlag_EndpointIdValid) };
+    Command::Command * ZCLcommand = mDevice->GetCommandSender();
+    TLV::TLVWriter writer  = ZCLcommand->CreateCommandDataElementTLVWriter();
+
+    TLV::TLVType dummyType = TLV::kTLVType_NotSpecified;
+
+    ReturnErrorOnFailure(writer.StartContainer(TLV::AnonymousTag, TLV::kTLVType_Structure, dummyType));
```

Comment:
Can we use a macro or something else for both the footer and the header ? The only thing that changes here is some ids that can be passed as parameters. It would make the generated code easier to read for humans.

Reply:
I don't want to use macros in this part.
These lines are not bounded logically, for example, line 27, 28 gets the command object and then the writer object, line 30 and 32 creates a struct in tlv. The spacing in generated files and the loop tag in template should be enough.